yarn-project/noir-contracts/src/contracts/token_bridge_contract/src/main.nr

```diff
@@ -0,0 +1,169 @@
+mod util;
+mod token_interface;
+
+// Minimal implementation of the token bridge that can move funds between L1 <> L2.
+// The bridge has a corresponding Portal contract on L1 that it is attached to 
+// And corresponds to a Token on L2 that uses the `AuthWit` accounts pattern.
+// Bridge has to be set as a minter on the token before it can be sued
+
+contract TokenBridge {
+    use dep::aztec::{
+        context::{Context},
+        state_vars::{public_state::PublicState},
+        types::type_serialisation::field_serialisation::{
+            FieldSerialisationMethods, FIELD_SERIALISED_LEN,
+        },
+        types::address::{AztecAddress, EthereumAddress},
+        oracle::compute_selector::compute_selector,
+    };
+
+    use crate::token_interface::Token;
+    use crate::util::{get_mint_content_hash, get_withdraw_content_hash, compute_secret_hash};
+
+    // Storage structure, containing all storage, and specifying what slots they use.
+    struct Storage {
+        token: PublicState<Field, 1>,
+    }
+
+    impl Storage {
+        fn init(context: Context) -> pub Self {
+            Storage {
+                token: PublicState::new(
+                    context,
+                    1,
+                    FieldSerialisationMethods,
+                ),
+            }
+        }
+    }
+
+    // Constructs the contract.
+    #[aztec(private)]
+    fn constructor() {
+        // Currently not possible to execute public calls from constructor as code not yet available to sequencer.
+        // let selector = compute_selector("_initialize((Field))");
+        // let _callStackItem = context.call_public_function(context.this_address(), selector, [context.msg_sender()]);
+    }
+
+    // Consumes a L1->L2 message and calls the token contract to mint the appropriate amount publicly
+    #[aztec(public)]
+    fn deposit_public(
```

Comment:
The naming with deposit and withdraw can make it quite hard to figure out the direction. Like, am I depositing into L1 from this contract, or am I depositing into L2 using a prior deposit.

Think it would make sense to make it a bit more clear that this is the second part of a deposit as well. 

Withdraw might become `withdrawToL1` or `exitToL1` or the like, and deposits becoming a claim of funds?